.github/workflows/homebrew-core-head.yml

```diff
@@ -25,6 +25,8 @@ jobs:
         # This is sub-optimal - our workflows shouldn't have to conform to their environment.
         # However, on the runner side, we can't hook into the workflow run to clean up after.
         run: brew uninstall semgrep || true
```

Comment:
Why do we need to do that? Why the self-hosted does not clean things by itself for each new job,
like what GHA does.

Reply:
https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#differences-between-github-hosted-and-self-hosted-runners

Reply:
We're using bare-metal instances, whereas the GHA runners are a single-use VM that starts fresh each time. We don't have tooling/processes for spinning up new instances for each job. It's a know limitation and on our radar.

Reply:
great link chris, would be nice to add it next to the uninstall step, so people understand why
we need those extra steps.